Makefile

```diff
@@ -3,62 +3,132 @@
 #                                                         :::      ::::::::    #
 #    Makefile                                           :+:      :+:    :+:    #
 #                                                     +:+ +:+         +:+      #
-#    By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+         #
+#    By: jodufour <jodufour@student.42.fr>          +#+  +:+       +#+         #
 #                                                 +#+#+#+#+#+   +#+            #
 #    Created: 2023/11/24 19:29:43 by mcourtoi          #+#    #+#              #
-#    Updated: 2024/03/01 22:03:28 by mcourtoi         ###   ########.fr        #
+#    Updated: 2024/03/06 02:23:58 by jodufour         ###   ########.fr        #
 #                                                                              #
 # **************************************************************************** #
 
 ######################################
 #              COMMANDS              #
 ######################################
-   AR = ${shell which ar} rcs
-  CXX = ${shell which c++}
- LINK = ${shell which clang++}
-MKDIR =	${shell which mkdir} -p
-   RM = ${shell which rm} -rf
+   AR := ${shell which ar} rcs
+  CXX := ${shell which c++}
+ LINK := ${shell which clang++}
+MKDIR := ${shell which mkdir} -p
+   RM := ${shell which rm} -rf
 
 ######################################
 #             EXECUTABLE             #
 ######################################
-NAME = ft_irc
+NAME := ft_irc
 
 #######################################
 #               LIBRARY               #
 #######################################
-LIB = lib${NAME}.a
+LIB := lib${NAME}.a
 
 #######################################
 #             DIRECTORIES             #
 #######################################
-SRC_DIR = src
-OBJ_DIR = obj
-PRV_DIR = private
-LIB_DIR = lib
-INC_DIR = include
+SRC_DIR := src
+OBJ_DIR := obj
+PRV_DIR := private
+LIB_DIR := lib
+INC_DIR := include
 
 ######################################
 #            SOURCE FILES            #
 ######################################
-SRC = \
+SRC := \
 	${addsuffix .cpp, \
 		main \
 	}
 
-LIB_SRC = \
+LIB_SRC := \
 	${addsuffix .cpp, \
 		${addprefix ${LIB_DIR}/, \
 			${addprefix class/, \
+				${addprefix Channel/, \
+					core \
+					methods \
+				} \
 				${addprefix Client/, \
-					Client \
 					core \
+					methods \
 				} \
-				${addprefix Channel/, \
-					Channel \
+				${addprefix exception/, \
```

Comment:
Because we recently talked about how we shall organize the files related to our classes, and we both agreed to make, for every class, a folder named the same as the class itself, and to put the constructors + destructor + accessors + mutators of the class in a first file named `core.cpp` inside that folder, and to put the rest of the methods of the class in a second file named `methods.cpp` inside this folder too.
And our exceptions are classes.